internal/proxy/task_query.go

```diff
@@ -543,22 +542,29 @@ func reduceRetrieveResults(ctx context.Context, retrieveResults []*internalpb.Re
 	idSet := make(map[interface{}]struct{})
 	cursors := make([]int64, len(validRetrieveResults))
 
+	realLimit := typeutil.Unlimited
 	if queryParams != nil && queryParams.limit != typeutil.Unlimited {
-		loopEnd = int(queryParams.limit)
-
+		realLimit = queryParams.limit
+		if !queryParams.reduceStopForBest {
+			loopEnd = int(queryParams.limit)
+		}
 		if queryParams.offset > 0 {
 			for i := int64(0); i < queryParams.offset; i++ {
-				sel := typeutil.SelectMinPK(validRetrieveResults, cursors)
+				sel := typeutil.SelectMinPK(validRetrieveResults, cursors, queryParams.reduceStopForBest, realLimit)
 				if sel == -1 {
 					return ret, nil
 				}
 				cursors[sel]++
 			}
 		}
 	}
+	reduceStopForBest := false
+	if queryParams != nil {
+		reduceStopForBest = queryParams.reduceStopForBest
+	}
 
 	for j := 0; j < loopEnd; j++ {
```

Comment:
The code should be the same as querynodev2/segments/results.go to remove the smaller ts row when two PKs are identical.

But it seems two places are not the same, breaking our designment of "get the latest when PKs are identical" 

Please help add them back at L578